### ISSUE TITLE
[BUG] Count in toggleterminal window adds 2

### ISSUE THREAD

**Issue by schemar** [akinsho/toggleterm.nvim#432]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Prerequisite: I mapped `<leader>w` to `:ToggleTerm<CR>` in `normal` mode.

Given that I am in a `toggleterm` buffer in `normal` mode
When I do `5<leader>w`
Then it opens the terminal with ID `7`

Same for all `count`s

### Expected Behavior

Prerequisite: I mapped `<leader>w` to `:ToggleTerm<CR>` in `normal` mode.

Given that I am in a `toggleterm` buffer in `normal` mode
When I do `5<leader>w`
Then it should open the terminal with ID `5`

Same for all `count`s

### Steps To Reproduce

1. Create key mapping to toggle terminals: `<leader>w`
2. Open the first terminal with `<leader>w` in `normal` mode
3. Open terminal `5` by doing `5<leader>w` in `normal` mode inside terminal `1`
4. Now terminal `7` is open and not terminal `5`

### Environment

```Markdown
- OS: macOS 13.3.1
- neovim version: 0.9.0
- Shell: zsh
```

### Anything else?

I already found out that the count is `2` too high for all mapped commands in a `toggleterm` buffer. Not only when opening another ToggleTerm window.

This error does not occur in neovim `terminal` buffers (or any other bufferes), only in `toggleterm` buffers.

When I change `./lua/toggleterm.lua` line `375` to also print the `count`, it already prints two too high:

```lua
  cmd("ToggleTerm", function(opts)
    print("count:", opts.count) -- Prints two more than I prefixed
    M.toggle_command(opts.args, opts.count)
  end, { count = true, complete = commandline.toggle_term_complete, nargs = "*" })
```

Let me know if you need anything else that I can support you with 💚 Thank you for this fantastic plugin!

### Further Research (can be ignored)

I created this test user command and bound it to `<leader>x`:
```lua
vim.api.nvim_create_user_command("SchemarTest", function(opts)
	print("test:", opts.count)
end, { count = true, nargs = "*" })
vim.cmd([[:noremap <leader>x :SchemarTest<CR>]])
```

This command consistently prints `count+2` instead of `count`.

**Comment by akinsho** [comment]
Hi @schemar,

So tbh I'm  not entirely sure I understand this issue very well but it seems like there's been some misunderstanding at some point.

`5:ToggleTerm<CR>` isn't a command I'd expect to work I'm not sure if it's a typo. In either case opening the correct terminal should be done with `:5ToggleTerm` which is how to add the range but if you are mapping the command, which you really don't need to because this plugin already includes an `open_mapping` because creating the mapping yourself is actually a little complicated, doing `5<leader>l` to a command that maps directly to `:ToggleTerm` won't work.

Basically It looks like are trying to wrap the command that this plugin already creates with another command that won't work but I'm not sure why bother wrapping the command.

The way to use this plugin is to set an open mapping and then you can do `<count><open-mapping>` e.g. `5<leader>w` but the point is that you should use the open mapping. Otherwise you can use the command but there's no point trying to wrap it because the command was already created to take a range like `3ToggleTerm`

**Comment by akinsho** [comment]
You can look at https://github.com/akinsho/toggleterm.nvim#setup for details on the command and how you *could* try to set it up but again I really don't recommend this unless it's something you're really familiar with

**Comment by schemar** [comment]
Sorry for the confusion and thank you for your patience.

I edited the description to be easier to understand (I hope).

In essence the bug comes down to this:

1. Map `<leader>w` to `:ToggleTerm<CR>`
2. Type `<leader>w` in normal mode in "normal" buffer
3. Change to `normal` mode inside ToggleTerm buffer
4. Type `3<leader>w`
5. It opens a ToggleTerm with ID `5` instead of `3`

Other commands also add `2` to the count.

**Comment by schemar** [comment]
It works as expected with `open_mapping`. I will use that now instead of using my own mapping. Thank you so much for your help!

**Comment by schemar** [comment]
Still, other mapped commands with a `count` add `2`, but I don't think that's an issue for me.

**Comment by akinsho** [comment]
> Still, other mapped commands with a count add 2

@schemar which commands do you mean? they might not be intended to take a count

**Comment by schemar** [comment]
> > Still, other mapped commands with a count add 2
> 
> @schemar which commands do you mean? they might not be intended to take a count

Granted, I only tested it with my own user command that I created for testing (see original description). Movements like `b`, `w`, etc. work as expected.
